internal/syncd/integrations/ecr/reconciler.go

```diff
@@ -2,27 +2,78 @@ package ecr
 
 import (
 	"context"
-
+	"fmt"
+	"path/filepath"
 	"ship-it/internal"
 
-	"github.com/google/go-github/github"
+	"github.com/google/go-github/v26/github"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/types"
 )
 
 type RepositoriesService interface {
-	GetContents(ctx context.Context, org, repo, path string, opts *github.RepositoryContentGetOptions) (*github.RepositoryContent, []*github.RepositoryContent, *github.Response, error)
-	UpdateFile(ctx context.Context, owner, repo, path string, opt *github.RepositoryContentFileOptions) (*github.RepositoryContentResponse, *github.Response, error)
+	UpdateFile(msg string, branch string, path string, fileContent []byte) (*github.RepositoryContentResponse, error)
+	GetFile(branch string, path string) (string, error)
+}
+
+type IndexerService interface {
+	Lookup(repo string) ([]types.NamespacedName, error)
 }
 
 type ImageReconciler struct {
-	// TODO
+	Org               string
+	RegistryChartPath string
+	Branch            string
+	RepoService       RepositoriesService
+	IndexService      IndexerService
 }
 
-func NewReconciler(org string, r RepositoriesService) *ImageReconciler {
+func NewReconciler(org string, prefix string, branch string, r RepositoriesService, i IndexerService) *ImageReconciler {
 	return &ImageReconciler{
-		// TODO
+		Org:               org,
+		RegistryChartPath: prefix,
+		Branch:            branch,
+		RepoService:       r,
+		IndexService:      i,
 	}
 }
 
+func updateResource(r *ImageReconciler, image *internal.Image, name string) error {
+	path := filepath.Join(r.RegistryChartPath, name+".yaml")
+	resourceStr, err := r.RepoService.GetFile(r.Branch, path)
+	if err != nil {
+		return errors.Wrapf(err, "failed to download custom resource file for path: %s", path)
+	}
+
+	rls := make(map[string]interface{})
+
+	err = yaml.Unmarshal([]byte(resourceStr), rls)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse the YAML file")
+	}
+
+	updatedRls := internal.WithImage(*image, rls)
```

Comment:
You are correct, I set the parameter to be a `map[string]interface{}` to get around the `runtime.RawExtension` type issues we were having earlier. Since the function does not depend on the `HelmRelease` type, the function is greatly simplified by using a generic `map[string]interface{}` and the standard `yaml` package.